File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict
def solution(clothes):
    answer = 1
    clothesDict = defaultdict(int)
    
    for name,clothesType in clothes:
        clothesDict[clothesType] += 1
    
    for key,value in clothesDict.items():
        answer *= (value + 1)
        
        
    
    
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        Map<String, Integer> clothMap = new HashMap<>();
        for(String[] c : clothes){
            clothMap.put(c[1], clothMap.getOrDefault(c[1],0) + 1);
        }
        
        for(String key : clothMap.keySet()){
            answer *= (clothMap.get(key) + 1);
        }
        
        return answer-1;
    }
}